azavea/district-builder-dtl-pa
# Add value choices checkboxes to submission form

## Overview

Adds a set of checkboxes to the submission form to require users to choose the values they prioritized when they created their plan.

### Checklist


### Demo

<img width="520" alt="screen shot 2018-08-28 at 8 40 11 pm" src="https://user-images.githubusercontent.com/447977/44737285-9b626000-ab02-11e8-97f4-8ebc65707338.png">

### Notes

- The acceptance criteria specify a dropdown, but that doesn't really make sense in a multi-select situation without creating custom UI elements. I opted to use a series of checkboxes instead.
-  It would have been nice to disable the submit button when too many / not enough checkboxes are selected, but that's not the existing functionality and the first thing I tried didn't work, so I skipped it.

## Testing Instructions

 * Create a valid plan (csv export attached if you need it)
[testnew.zip](https://github.com/azavea/district-builder-dtl-pa/files/2328961/testnew.zip)

 * Start checking boxes and confirm you get an error if you select 0 or more than 3.
 * Confirm that submitting the form doesn't work if the error is displayed.
 * Submit a form and then run `./manage.py submission_report 1` (unless you've made other submissions since the last time you wiped your database, in which case open a `./manage.py dbshell` and use `SELECT id FROM redistricting_plansubmission;` to find the highest ID and use that).
 * Confirm that the submission report contains the criteria you selected when you submitted the form.
 * Confirm that the acceptance criteria (including comments) in https://www.pivotaltracker.com/story/show/159772749 are met.
 * Confirm that leaving a required field (other than the checkboxes) blank highlights it with a red border.

Closes #159772749
Closes #159772649
